iothub/device/src/Transport/Amqp/AmqpUnit.cs

```diff
@@ -123,30 +131,27 @@ internal async Task<AmqpIotSession> EnsureSessionIsOpenAsync(CancellationToken c
 
             try
             {
-                if (_amqpIotSession == null
-                    || _amqpIotSession.IsClosing())
+                if (_amqpIotSession == null || _amqpIotSession.IsClosing())
                 {
                     _amqpIotSession?.SafeClose();
```

Comment:
Discussed offline - adding locks or checks for this isn't possible because the AMQP library doesn't provide any callbacks for notifying us of the state. Instead, we have error handling logic when we open links or try to perform operations on opened links. If the operation throws an exception, the error handling code will determine if it is to be tried, and it will retry, if necessary.